src/endpoints/authentication/jwt_auth.cpp

```diff
@@ -96,10 +186,17 @@ namespace ccf
               time_now,
               token.payload_typed.exp);
           }
+          else if (
+            issuers &&
+            !validate_issuers(token, *issuers, std::ref(validated_issuer)))
+          {
+            error_reason =
+              fmt::format("Kid {} failed issuer validation", key_id);
+          }
           else
           {
             auto identity = std::make_unique<JwtAuthnIdentity>();
-            identity->key_issuer = key_issuers->get(key_id).value();
+            identity->key_issuer = validated_issuer;
```

Comment:
We never use the `Legacy::JWT_PUBLIC_SIGNING_KEY_ISSUER` table, which means (if I'm reading right) that this `validated_issuer` string will still be empty, if `issuers` is empty and we went through the `fallback_keys` path? Can we add a path in the fallback case above, something like:

```
if (!issuers)
{
  auto fallback_issuer = tx.ro<...>(ccf::Tables::Legacy::JWT_PUBLIC_SIGNING_KEY_ISSUER);
  validated_issuer = fallback_issuer->get(key_id).value_or({});
}
```

Or maybe we set `issuers` to a single-entry vector containing the value from the legacy map, and go through this validation path? Whatever makes sense, but I think we should try harder to use the fallback values if we're doing any fallback handling here.

Reply:
Good catch. Will update with a fallback value, to preserve the old behaviour.

Reply:
Done